### ISSUE TITLE
Documentation doesn't cover MWE using launch.py script

### ISSUE THREAD

**Issue by kirk86** [pytorch/pytorch#35160]
## 📚 Documentation

Documentation related to `torch.nn.parallel.DistributedDataParallel()` don't include any examples of how to properly use it in combination with `launch.py`

The screenshot below from `launch.py` shows one can call their script e.g. `main.py` using `launch.py` but gives no proper example on how to set up `main.py` in order to work correctly with `launch.py`

In order not to repeat myself there's a longer discussion in the [forums](https://discuss.pytorch.org/t/distributed-parallel-one-machine-multi-gpu-multi-process/72177)

Also the claim that in general `torch.nn.parallel.DistributedDataParallel()` is faster than `torch.nn.parallel.DataParallel()`, is not completely true, at least not in my experiments on a single node multi-gpu (i.e. 2 gpus).

This needs verification.

If that's the case then docs should reflect that and state that `DistributedDataParallel` is faster only when you have n-nodes or n-gpus?

![image](https://user-images.githubusercontent.com/2902390/77232091-e6394200-6b96-11ea-9a67-97614701766d.png)

**Comment by mrshenli** [comment]
@kirk86 could you please help to check if https://github.com/pytorch/examples/pull/74 is sufficient to close this issue? Thanks!

**Comment by kirk86** [comment]
@mrshenli Unless I'm missing sth, I think the issue you are referring to has to do with resuming checkpoints and continuing training. The issue that I'm referring has to do with the `lanuch.py` script from [distributed](https://github.com/pytorch/pytorch/blob/f326045b3757236aabe367dfca1894be14ce31ef/torch/distributed/launch.py) module. 

The purpose of this script is to spawn multiple processes automatically instead of having to do it manually like in the [imagenet example](https://github.com/pytorch/examples/blob/master/imagenet/main.py)

Essentially what I would like to see is the following, shown in the screenshot below, from imagenet example referenced above, but for `launch.py`.

Either someone plz verity that `launch.py` works and thus provide a concrete example for others to follow and how to properly use it or let's not confuse ppl and ditch it altogether and update the docs to say that if you want distributed training you need to setup and spawn multiple processes manually like in the imagenet example above.

![image](https://user-images.githubusercontent.com/2902390/77585908-30932980-6edd-11ea-8a09-54b9e037c41e.png)

**Comment by mrshenli** [comment]
@kirk86 sorry wrong link. I missed a 3 at the end of the URL. Just updated the pointer.

**Comment by kirk86** [comment]
@mrshenli Yes that's exactly what was missing :smiley:. That will do, thanks!
